mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -500,6 +500,15 @@ async fn run_db_migration_impl(ctx: &MmArc, db_id: Option<&str>, shared_db_id: O
 }
 
 pub async fn lp_init_continue(ctx: MmArc) -> MmInitResult<()> {
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        let dbdir = ctx.dbdir(None);
+        fs::create_dir_all(&dbdir).map_to_mm(|e| MmInitError::ErrorCreatingDbDir {
+            path: dbdir.clone(),
+            error: e.to_string(),
+        })?;
+    }
```

Comment:
This should be before `initialize_wallet_passphrase` as it was done before https://github.com/KomodoPlatform/komodo-defi-framework/blob/1675a53752aaa5d3a01a5f88566592b42264f990/mm2src/mm2_main/src/lp_native_dex.rs#L553-L554
You don't need to create the default db dir, only the global one. We probably also need to `ensure_dir_is_writable`, just in case.